pkg/cmd/command.go

```diff
@@ -21,99 +28,193 @@ func rootCommand() *cobra.Command {
 
 func createCommand() *cobra.Command {
 	command := &cobra.Command{
-		Use: "create",
-	}
-
-	command.AddCommand(createTcpServerCommand())
-	command.AddCommand(createHttpServerCommand())
-
-	return command
-}
-
-func createHttpServerCommand() *cobra.Command {
-	command := &cobra.Command{
-		Use:          "http",
+		Use:          "create",
 		SilenceUsage: true,
 	}
 
+	var apiToken string
 	var name string
 	var org string
 	var region string
+	var tcp bool
+	var ports []string
 
 	command.Flags().StringVar(&name, "name", "", "")
-	command.Flags().StringVar(&org, "org", "personal", "")
-	command.Flags().StringVar(&region, "region", "", "")
+	command.Flags().StringVar(&org, "org", "", "")
+	command.Flags().StringVar(&region, "region", "ams", "")
+	command.Flags().StringVar(&apiToken, "api-token", "", "")
+	command.Flags().BoolVar(&tcp, "tcp", false, "")
+	command.Flags().StringSliceVar(&ports, "port", []string{}, "")
 
 	command.RunE = func(command *cobra.Command, args []string) error {
-		configuredPorts := []Ports{{InternalPort: 8000, ExternalPorts: []uint{80, 443}}}
+		inletsVersion := "0.9.9"
+
+		if apiToken == "" {
+			apiToken = os.Getenv("FLY_API_TOKEN")
+		}
+
+		if apiToken == "" {
+			fmt.Println("give a value --api-token or set the environment variable \"FLY_API_TOKEN\"")
+			return nil
+		}
+
+		if tcp && len(ports) == 0 {
+			return fmt.Errorf("at least one --port is required when creating a TCP exit node")
+		}
+
+		token, err := password.Generate(64, 10, 0, false, true)
+		if err != nil {
+			return err
+		}
 
 		if name == "" {
 			name = names.GetRandomName()
 		}
 
-		token, err := password.Generate(64, 10, 0, false, true)
+		hostReq := &cp.BasicHost{
+			Region: region,
+			Name:   name,
+			Additional: map[string]string{
+				"inlets-tcp":     fmt.Sprintf("%v", tcp),
+				"inlets-ports":   strings.Join(ports, ","),
+				"inlets-token":   token,
+				"inlets-version": inletsVersion,
+			},
+		}
+
+		provisioner, err := provision.NewFlyProvisioner(apiToken, org, region)
 		if err != nil {
 			return err
 		}
+		defer provisioner.Close()
 
-		app := App{
-			Mode:          "http",
-			InletsVersion: "0.9.1",
-			Name:          name,
-			Org:           org,
-			Region:        region,
-			Token:         token,
-			Ports:         configuredPorts,
+		hostRes, err := provisioner.Provision(*hostReq)
+		if err != nil {
+			return err
+		}
+
+		fmt.Printf("Host: %s, status: %s\n", hostRes.ID, hostRes.Status)
+
+		poll := time.Second * 2
+		max := 500
+		for i := 0; i < max; i++ {
+			time.Sleep(poll)
+
+			hostStatus, err := provisioner.Status(hostRes.ID)
+			if err != nil {
+				return err
+			}
+
+			fmt.Printf("[%d/%d] Host: %s, status: %s\n",
+				i+1, max, hostStatus.ID, hostStatus.Status)
+
+			if hostStatus.Status == "active" {
+				if tcp {
+					fmt.Printf(`inlets Pro TCP (%s) server summary:
```

Comment:
If this could be back-ported into the cloud-provision library, then perhaps it could be used from inletsctl and inlets-operator, making having to print custom messages etc redundant?